metrics.go

```diff
@@ -89,44 +89,27 @@ func NewSummaryWithLabels(opts MetricsOpts) (*prometheus.SummaryVec, error) {
 	}, opts.Labels), nil
 }
 
-// RegisterMetrics registers the basic Prometheus metrics and the well known strategies.
-func RegisterMetrics(metrics interface{}) error {
-	errPromType := errors.New("metric is not a Prometheus Collector")
-
-	switch v := metrics.(type) {
-	case *prometheus.CounterVec,
-		*prometheus.GaugeVec,
-		*prometheus.HistogramVec,
-		*prometheus.SummaryVec:
-		m, ok := v.(prometheus.Collector)
-		if !ok {
-			return errPromType
-		}
+// RegisterMetrics registers the complex metrics strategies, metrics built as
+// custom types using structs, like the well known strategies (USE/RED/FourGoldenSignals).
+func RegisterComplexMetrics(metrics interface{}) error {
+	if reflect.ValueOf(metrics).Kind() != reflect.Struct {
+		return errors.New("complex metric is not built with a stuct")
+	}
 
-		prometheus.MustRegister(m)
+	l := reflect.TypeOf(metrics).NumField()
+	if l < 1 {
+		return errors.New("metric strategies needs at least one field")
+	}
 
-	// registers the complex metrics strategies, metrics built as custom types
-	// using structs, like the well known strategies (USE/RED/FourGoldenSignals).
-	default:
-		if reflect.ValueOf(v).Kind() != reflect.Struct {
-			return errors.New("complex metric is not built with a stuct")
-		}
+	values := reflect.ValueOf(metrics)
 
-		l := reflect.TypeOf(v).NumField()
-		if l < 1 {
-			return errors.New("metric strategies needs at least one field")
+	for i := 0; i < l; i++ {
+		m, ok := values.Field(i).Interface().(prometheus.Collector)
+		if !ok {
+			return errors.New("metric is not a Prometheus Collector")
```

Comment:
Do we also want to add the field name in the error?

Reply:
@ajk314 I like this suggestion and will add it to a subsequent PR